orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/ExecutionLauncher.java

```diff
@@ -64,6 +75,36 @@ public T start(T execution) throws Exception {
     return execution;
   }
 
+  protected T handleStartupFailure(T execution, Throwable failure) {
+    final String canceledBy = "system";
+    final String reason = "Failed on startup: " + failure.getMessage();
+    final ExecutionStatus status = ExecutionStatus.TERMINAL;
+    executionRepository.updateStatus(execution.getId(), status);
+    Function<Execution, Execution> reloader;
+    String executionType;
+    if (execution instanceof Pipeline) {
+      executionType = "pipeline";
+      reloader = (e) -> executionRepository.retrievePipeline(e.getId());
+    } else if (execution instanceof Orchestration) {
+      executionType = "orchestration";
+      reloader = (e) -> executionRepository.retrieveOrchestration(e.getId());
+    } else {
+      log.error("Unknown execution type: " + execution.getClass().getSimpleName());
+      executionType = "unknown";
+      reloader = (e) -> {
+        e.setCancellationReason(reason);
+        e.setCanceled(true);
+        e.setCanceledBy(canceledBy);
+        e.setStatus(status);
+        return e;
+      };
```

Comment:
I get that but why is it then setting the fields that should have been set already?